internal/gengapic/genrest.go

```diff
@@ -838,12 +838,19 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	p("  return nil, err")
 	p("}")
 	p("")
-	p("unm := protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}")
-	p("rsp := &%s.%s{}", outSpec.Name, outType.GetName())
-	p("")
-	p("if err := unm.Unmarshal(buf, rsp); err != nil {")
-	p("  return nil, maybeUnknownEnum(err)")
-	p("}")
+	if outSpec.Name == "httpbodypb" && outType.GetName() == "HttpBody" {
```

Comment:
Can we instead compare the `outType`'s fully-qualified proto name to that of the `HttpBody`? `google.api.HttpBody`

You can build the FQN of the `outType` by retrieving the parent file and combining its package with the message name. `fqn := fmt.Sprintf("%s.%s", g.descInfo.ParentFile[outType].GetPackage(), outType.GetName())`. I realize this pattern doesn't work for nested messages, but this message isn't nested so it doesn't matter.

Reply:
Done.